components/brave_wallet_ui/components/buy-send-swap/buy/index.tsx

```diff
@@ -36,20 +48,51 @@ function Buy (props: Props) {
     defaultCurrencies,
     onInputChange,
     onSubmit,
-    onChangeBuyView
+    onChangeBuyView,
+    selectedBuyOption,
+    onSelectBuyOption,
+    isAvailableOnWyre,
+    isAvailableOnRamp
   } = props
+  const [buyOptions, setBuyOptions] = React.useState<BuyOption[]>(BuyOptions)
 
   const onShowAssets = () => {
     onChangeBuyView('assets', 'from')
   }
 
+  React.useEffect(() => {
+    if (isAvailableOnRamp && isAvailableOnWyre) {
+      setBuyOptions(BuyOptions)
+    }
+
+    if (isAvailableOnRamp && !isAvailableOnWyre) {
+      setBuyOptions(BuyOptions.filter(option => option.id !== BraveWallet.OnRampProvider.kWyre))
+    }
+
+    if (!isAvailableOnRamp && isAvailableOnWyre) {
+      setBuyOptions(BuyOptions.filter(option => option.id !== BraveWallet.OnRampProvider.kRamp))
+    }
+  }, [isAvailableOnWyre, isAvailableOnRamp])
```

Comment:
This will become quite complex if we add another onramp provider. Instead of `isAvailableOnWyre` and  `isAvailableOnRamp`, we can have a list of all onramp providers that allows buying the selected asset. In that case, the body of this whole `useEffect` is a single `filter`.